lib/go/thrift/http_client.go

```diff
@@ -197,11 +197,11 @@ func (p *THttpClient) Flush(ctx context.Context) error {
 	// Close any previous response body to avoid leaking connections.
 	p.closeResponse()
 
-	// Request might not have been fully read by http client.
-	// Reset so we don't send the remains on next call.
-	defer p.requestBuffer.Reset()
-
-	req, err := http.NewRequest("POST", p.url.String(), p.requestBuffer)
+	// Give up the ownership of the current request buffer to http request,
+	// and create a new buffer for the next request.
+	buf := p.requestBuffer
+	p.requestBuffer = new(bytes.Buffer)
+	req, err := http.NewRequest("POST", p.url.String(), buf)
```

Comment:
Now it looks like there is a memory leak here, since the buffer is not being`.Reset()`-ed anywhere.

But I don't think this is a problem, because it will be collected by the garbage collector.